Automattic/woocommerce-payments
# Accept charges in multiple currencies

Addresses "accept charges" aspect of #777 

**Do not merge** without 386-gh-woocommerce-payments-server and accommodations for currency in Payments dashboard.

#### Changes proposed in this Pull Request

Use order (instead of hard-coded) currency for payments.

<s>Still needs proper handling of [zero-decimal currencies](https://stripe.com/docs/currencies#zero-decimal) (i.e. [passing through](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/52761eaedf058a26a5e6e317b6cfa1de74f71211/includes/class-wc-stripe-helper.php#L179-L180) rounded amount instead of multiplying by 100).</s>

Should also test the error case when attempting to use an [unsupported currency](https://stripe.com/docs/currencies#presentment-currencies) (e.g. INR with American Express).

We currently have no tests for `prepare_amount` – should we add some in this PR to cover the added bit of complexity?

#### Testing instructions

- Install & activate [WooCommerce Multi-Currency](https://docs.woocommerce.com/document/multi-currency/) extension
- Configure with a service provider (e.g. OpenExchangeRates App ID) and a least one currency (e.g. ZAR and CLP)
- Go to the store front-end with currency specified (e.g. `/shop/?currency=ZAR`) and check out
- Verify that the transaction amount is reflective of the correct price regardless of presentment currency
- Check out again with a zero-decimal currency (e.g. `/shop/?currency=CLP`) and verify the transaction amount
- Ensure transactions listing contains entry for the payment (the used currency is one you have chosen for your Stripe account, not the order currency)

Update: Please use VND as currency, as it'll properly be displayed on the client-side.

-------------------
